internal/campaigns/types.go

```diff
@@ -486,11 +486,15 @@ func (cs Changesets) IDs() []int64 {
 	return ids
 }
 
-// IDs returns the RepoIDs of all changesets in the slice.
+// IDs returns the unique RepoIDs of all changesets in the slice.
 func (cs Changesets) RepoIDs() []api.RepoID {
-	repoIDs := make([]api.RepoID, len(cs))
-	for i, c := range cs {
-		repoIDs[i] = c.RepoID
+	repoIDMap := make(map[api.RepoID]struct{})
+	for _, c := range cs {
+		repoIDMap[c.RepoID] = struct{}{}
+	}
+	repoIDs := make([]api.RepoID, 0)
```

Comment:
`make([]foo, 0)` is equivalent to `[]foo{}` or `var s []foo`, so it doesn't make a lot of sense to use this form.

But since you know the capacity, you can do this:

```suggestion
	repoIDs := make([]api.RepoID, 0, len(repoIDMap))
```